Buddies.Web/src/pages/projects/[pid].tsx

```diff
@@ -0,0 +1,235 @@
+  Location: 'Unknown',
+  ProjectOwner: 'D.N.E',
+  ProjectOwnerEmail: 'D.N.E',
+  MaxMembers: 1,
+  Category: 'Not Found',
+  MemberLst: memberLst,
+  InvitedLst: [],
+};
+
+let projectId: string | string[] | undefined = '';
+let currId: number; // id of the user viewing the page
+const invitedIds: number[] = []; // list of userId's in the invited list
+const memberIds: number[] = []; // list of userId's in the member list
+let ownerId: number; // Id of the owner of the project
+
+/* Project profile page. Responsible for putting all the components that make up the
+  page together. It also sends GET requests to get a project by its id. And a
+  PUT request, to let invited users join the project.
+*/
+const Project: React.VFC = () => {
+  const [project, setProject] = React.useState<ProjectProfile>(defaultProject);
+  const authState = authStore((state: { authState: any; }) => state.authState);
+  const router = useRouter();
+
+  /* Gets project by id and then creates necessary global data structures.
+     (memberLst, invitedLst, project object)
+  */
+  function getProject() {
+    if (!(typeof projectId === 'string')) {
+      alert('error');
+      return;
+    }
+    axios.get(baseURL + projectId).then((res) => {
+      const newMemberLst = [];
+      for (let i = 0; i < res.data.members.length; i += 1) {
+        newMemberLst.push({
+          FirstName: res.data.members[i].firstName,
+          LastName: res.data.members[i].lastName,
+          UserId: res.data.members[i].userId,
+        });
+        memberIds.push(res.data.members[i].userId);
+        if (res.data.email === res.data.members[i].email) {
+          ownerId = res.data.members[i].userId;
+        }
+      }
+      const newInvitedLst = [];
+      for (let i = 0; i < res.data.invitedUsers.length; i += 1) {
+        newInvitedLst.push({
+          FirstName: res.data.invitedUsers[i].firstName,
+          LastName: res.data.invitedUsers[i].lastName,
+          UserId: res.data.invitedUsers[i].userId,
+        });
+        invitedIds.push(res.data.invitedUsers[i].userId);
+      }
+
+      const newProject = {
+        Title: res.data.title,
+        Description: res.data.description,
+        Location: res.data.location,
+        ProjectOwner: res.data.username,
+        ProjectOwnerEmail: res.data.email,
+        MaxMembers: res.data.maxMembers,
+        Category: res.data.category,
+        MemberLst: newMemberLst,
+        InvitedLst: newInvitedLst,
+      };
+      setProject(newProject);
+    }).catch((error) => {
+      alert(error);
+    });
+  }
+
+  const addMember = async () => {
+    if (!(typeof projectId === 'string')) {
+      alert('error');
+      return;
+    }
+    const res = await axios.post(`${baseURL + projectId}/join/`, currId).catch((error) => {
+      alert(error);
+    });
+    if (true && res) {
```

Comment:
Does res by itself not work?

Reply:
It does, I fixed it now. I have no clue why I had "true && res". I think, I had some typescript error and using "true && res" made it go away. My vs code might've just been lagging.